File: app/src/main/java/com/ilyap/yuta/utils/ProjectStatus.java
package com.ilyap.yuta.utils;

import lombok.RequiredArgsConstructor;
import org.jetbrains.annotations.NotNull;

import java.util.Arrays;

@RequiredArgsConstructor
public enum ProjectStatus {
    PROCESS("в работе"),
    PAUSE("приостановлен"),
    COMPLETE("завершен");

    private final String text;

    @Override
    public @NotNull String toString() {
        return text;
    }

    public static ProjectStatus fromText(String text) throws IllegalArgumentException {
        return Arrays.stream(ProjectStatus.values())
                .filter(v -> v.toString().equals(text))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("unknown value: " + text));
    }
}

File: domain/src/main/java/com/yuta/domain/model/ProjectStatus.kt
package com.yuta.domain.model

enum class ProjectStatus(private val text: String) {

    PROCESS("в работе"),
    PAUSE("приостановлен"),
    COMPLETE("завершен");

    override fun toString(): String = text

    companion object {
        fun fromText(text: String): ProjectStatus {
            return entries.find { it.text == text }
                ?: throw IllegalArgumentException("Unknown value: $text")
        }
    }
}
